File: USACO/gift1/gift1.py
"""
ID: felixaz1
LANG: PYTHON3
TASK: gift1
"""
fin = open("gift1.in", 'r')
fout = open('gift1.out', 'w')

n_of_names = int(fin.readline().rstrip())
people = {fin.readline().rstrip(): 0 for _ in range(n_of_names)}
for _ in range(n_of_names):
    person = fin.readline().rstrip()
    money, n_of_people = map(int, fin.readline().rstrip().split())
    for _ in range(n_of_people):
        people[fin.readline().rstrip()] += money // n_of_people
    people[person] -= money - (money % n_of_people) if n_of_people != 0 else 0

for name, money in people.items():
    fout.write(name + " " + str(money) + '\n')

File: USACO/gift1/gift1.cpp
/*
ID: felixaz1
LANG: C++11
TASK: gift1
*/

#include <fstream>
#include <bits/stdc++.h>

using namespace std;

#define forn(s, e) for(int i = s; i < e; i++)

int main () {
    ifstream in("gift1.in");
    ofstream out("gift1.out");

    int n_of_names; in >> n_of_names;
    map<string, int> people;
    vector<string> order;
    string person;
    forn(0, n_of_names) { in >> person; people[person] = 0; order.push_back(person); }
    forn(0, n_of_names) {
        in >> person;
        int money, n_of_people;
        in >> money >> n_of_people;
        string reciever;
        forn(0, n_of_people) {
            in >> reciever;
            people[reciever] += money / n_of_people;
        }
        if (n_of_people != 0) people[person] -= money - (money % n_of_people);
    }
    for_each(order.begin(), order.end(), [&out, &people](string s) {out << s << " " << people[s] << endl; } );

}